tests/e2e/default/sms/africas-talking.wdio-spec.js

```diff
@@ -0,0 +1,252 @@
+const utils = require('../../../utils');
+const querystring = require('querystring');
+const commonPage = require('../../../page-objects/default/common/common.wdio.page');
+const messagesPage = require('../../../page-objects/default/sms/messages.wdio.page');
+const loginPage = require('../../../page-objects/default/login/login.wdio.page');
+const reportsPage = require('../../../page-objects/default/reports/reports.wdio.page');
+
+const CREDENTIAL_PASS = 'yabbadabbadoo';
+const CREDENTIAL_KEY = 'africastalking.com:incoming';
+
+describe('africas talking api', () => {
+  before( () => utils.saveCredentials(CREDENTIAL_KEY, CREDENTIAL_PASS) );
+
+  describe('- gateway submits new WT sms messages', () => {
+    const submitSms = body => {
+      const content = querystring.stringify(body);
+      return utils.request({
+        method: 'POST',
+        path: `/api/v1/sms/africastalking/incoming-messages?key=${CREDENTIAL_PASS}`,
+        body: content,
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+          'Content-Length': content.length,
+        },
+        json: false,
+      });
+    };
+
+    beforeEach(async () => {
+      submitSms({ from: '+64271234567', text: 'Hello', id: 'messageID' });
+      await loginPage.cookieLogin();
+    });
+
+    it('- shows content', async () => {
+      const rawNumber = '+64271234567';
+      const message = 'Hello';
+
+      await commonPage.goToMessages();
+      const { heading, summary} = await messagesPage.getMessageInListDetails(rawNumber);
+      expect(heading).to.equal(rawNumber);
+      expect(summary).to.equal(message);
+
+      await messagesPage.openMessage(rawNumber);
+
+      const { name } = await messagesPage.getMessageHeader();
+      expect(name).to.equal(rawNumber);
+
+      const { content, state, dataID } = await messagesPage.getMessageContent(1);
+      expect(content).to.equal(message);
+      expect(state).to.equal('received');
+
+      const doc = await utils.getDoc(dataID);
+      expect(doc.sms_message && doc.sms_message.gateway_ref).to.equal('messageID');
+    });
+
+  });
+
+  describe('- gateway submits WT sms status updates', () => {
+    let savedDoc;
+    const messageGatewayRef1 = 'f0f237ab-dd34-44a8-9f17-caaa022be947';
+    const messageGatewayRef2 = 'f0cb5078-57da-427c-b3a9-b76ae581e5da';
+    const messageGatewayRef3 = 'f21a9fe4-2da0-49c1-a0cf-13f2554d7430';
+    const messageTo1 = '+64275555556';
+    const messageTo2 = '+64275555557';
+    const messageTo3 = '+64275555558';
+    const messageContent1 = 'Thank you for registering Shannon. Their pregnancy ID is 28551, and EDD is ' +
+      'Sun, Dec 18th, 2016';
+    const messageContent2 = 'Please remind Shannon (28551) to visit the health facility for ANC visit this week. ' +
+      'When she does let us know with "V 28551". Thanks!';
+    const messageContent3 = 'Did Shannon attend her ANC visit? When she does, respond with "V 28551". Thank you!';
+
+    const report = {
```

Comment:
What do you think about using [sms-pregnancy](https://github.com/medic/cht-core/blob/master/tests/factories/cht/reports/sms-pregnancy.js) factory? You may need to change the original factory and specify [options](https://github.com/rosiejs/rosie#programmatic-generation-of-attributes) to generate the attributes.
As an example, I use it in a couple of factories [here](https://github.com/medic/cht-core/tree/master/tests/factories/brac).

Reply:
That was a great idea Lore, thank you for helping in the process! 🤍